org.eclipse.dawnsci.analysis.api/src/org/eclipse/dawnsci/analysis/api/persistence/IPersistentFile.java

```diff
@@ -107,7 +109,38 @@
 	 * @throws Exception 
 	 */
 	public void setData(IDataset data) throws Exception;
+
 	
+	/**
+	 * Write data and axes in a single call.
+	 * 
+	 * For 1D data set yAxis as null
+	 * 
+	 * @param data
+	 * @param xAxis
+	 * @param yAxis
+	 * @throws Exception
+	 */
+	public void setData(IDataset data, IDataset xAxis, IDataset yAxis) throws Exception;
+
+	/**
+	 * Write data and axes in a single call.
+	 * <p>
+	 * <b>Important:</b> the axes should be in order of the dimensions of the dataset
+	 * @param data
+	 * @param axes
+	 * @throws Exception
+	 */
+	default public void setData(IDataset data, IDataset... axes) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)